filters/orientationinterpreter/orientationinterpreter.cpp

```diff
@@ -120,8 +120,8 @@ void OrientationInterpreter::accDataAvailable(unsigned, const AccelerationData*
 
 bool OrientationInterpreter::overFlowCheck()
 {
-    int vector = ((data.x_ * data.x_ + data.y_ * data.y_ + data.z_ * data.z_) / 1000);
-    return !((vector >= minLimit) && (vector <= maxLimit));
+    int m = int(sqrtf(data.x_ * data.x_ + data.y_ * data.y_ + data.z_ * data.z_));
+    return m < minLimit || m > maxLimit;
```

Comment:
Would be faster to drop `sqrtf()` and compare against (cached) squared limits?

Reply:
hmm, that might make sense - keep to good part of old logic

Reply:
Done as suggested + tried to lessen chances of numerical overflows causing problems.